pandas/core/frame.py

```diff
@@ -8764,11 +8764,22 @@ def update(
         if not isinstance(other, DataFrame):
             other = DataFrame(other)
 
-        other = other.reindex(self.index)
+        if other.index.has_duplicates:
+            raise ValueError("Update not allowed with duplicate indexes on other.")
```

Comment:
Can you add a note that this is not supported to the docstring. There isn't currently a notes section, something like this:

https://github.com/pandas-dev/pandas/blob/737d390381ffd8006f15d738c934acb659b5c444/pandas/core/frame.py#L1450-L1454

Reply:
@rhshadrach Done. 

`DataFrame.reindex` raises on the same condition and does not mention it on the docs. What about adding a similar note there?

Reply:
Slight preference on keeping this PR just modifying update, but no strong objection to modifying the reindex docs.

Reply:
Ok. It can be done in another PR.